src/modules/aws_codedeploy/mappers/deployment.ts

```diff
@@ -174,20 +128,23 @@ export class CodedeployDeploymentMapper extends MapperBase<CodedeployDeployment>
       }
     },
     updateOrReplace: (_a: CodedeployDeployment, _b: CodedeployDeployment) => 'update',
-    update: async (deployments: CodedeployDeployment[], ctx: Context) => {
+    update: async (es: CodedeployDeployment[], ctx: Context) => {
+      // just restore the values for the updated records
       const out = [];
-      for (const deployment of deployments) {
-        if (!deployment.application || !deployment.deploymentGroup) continue; // cannot update a deployment group without app or id
-
-        const cloudRecord = ctx?.memo?.cloud?.CodedeployDeployment?.[this.entityId(deployment)];
-        cloudRecord.id = deployment.id;
-        cloudRecord.deploymentGroup = deployment.deploymentGroup;
-        await this.module.deployment.db.update(cloudRecord, ctx);
-        out.push(cloudRecord);
+      for (const e of es) {
+        if (e.deploymentId) {
+          const depId = e.deploymentId;
+          const region = e.region;
+          const cloudRecord = ctx?.memo?.cloud?.Deployment?.[this.generateId({ depId, region })];
+          cloudRecord.id = e.id;
+          await this.module.deployment.db.update(cloudRecord, ctx);
+          out.push(cloudRecord);
+        }
       }
       return out;
     },
     delete: async (deployments: CodedeployDeployment[], ctx: Context) => {
+      // deployments can't actually be deleted, we just need to recreate
```

Comment:
do they have an expiration date or AWS deleted them at some point?

Reply:
they are deleted when you delete the related application. That's what i've seen from the user interface. You cannot delete deployments per se, but they get removed when you delete the main app.